### DISCUSSION TITLE
Code documentation is sparse - is it on purpose?

### DISCUSSION THREAD

**Discussion by marcinjahn** [gtk-rs/gtk-rs-core#1396] [General]
The crates have rather limited amount of docs. When one enters something like [glib crate](https://docs.rs/glib/0.19.4/glib/index.html),  it contains some intro, but that seems to be pretty much it. The types themselves, their methods and props are rather undocumented.

Is it on purpose, with an expectation of the C API documentation being the place to go to, or is it just something missing at the moment, but desirable to have at some point?

**Comment by bilelmoussaoui** [comment]
You should use the docs from https://gtk-rs.org/gtk-rs-core/. The docs can't be embedded into the Rust files for licensing reasons (MIT vs LGPL)

**Reply by marcinjahn** [reply]
Interesting... and sad actually. Thanks for this pointer anyway, it's really useful

**Reply by bilelmoussaoui** [reply]
In an ideal world, docs.rs would redirect to the doc link used in the Cargo.toml instead of pointing to mostly empty docs. I am sure other crates have this issue

**Reply by sdroege** [reply]
The windows crate has basically the same issue, and this is how they solved it: https://docs.rs/windows/latest/windows/
